### PR TITLE
fix: Remove LINQ usage in PassModifierKeys

### PR DESCRIPTION
**Pull request by m3taphysics**
Before:
Before_PassModifierKeys_No_Linq2

After:
After_PassModifierKeys_No_Linq

### Testing
- [ ] Ensure input modifier keys still work as expected

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using UnityEngine;
 using UnityEngine.EventSystems;
 using DCL.Configuration;
+using UnityEngine.Profiling;
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs:8]: Nitpickin here, but I dont think we need this.

**Review by AjimenezDCL — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs [hunk 2]
```diff
@@ -631,7 +632,16 @@ public static Boolean PassModifierKeys(KeyCode[] modifierKeys)
...
+                // Very confusing function, but this should avoid allocations and LINQ
+                var contains = false;
+                foreach (KeyCode kc in modifierKeys)
+                {
+                    if (kc == keyCode)
+                    {
+                        contains = true;
+                    }
+                }
+                if (contains != pressed)
```
Comment (eordano) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs:644]: Another nitpick for readability

```suggestion
                var anyModifierKeysActive = false;
                foreach (KeyCode kc in modifierKeys)
                {
                    if (kc == keyCode)
                    {
                        anyModifierKeysActive = true;
                        break;
                    }
                }
                if (anyModifierKeysActive != pressed)
```

**Review by lorux0 — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs [hunk 3]
```diff
@@ -631,7 +632,16 @@ public static Boolean PassModifierKeys(KeyCode[] modifierKeys)
             }
             else
             {
-                if (modifierKeys.Contains(keyCode) != pressed)
+                // Very confusing function, but this should avoid allocations and LINQ
+                var contains = false;
```
Comment (popuz) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs:636]: it was done in [this PR](https://github.com/decentraland/unity-renderer/pull/4186/files#diff-802617846bec437693e2fb4e126ce3ce8ff3777de96f79a31625d9b5a7ded0c8) as well (under review)

**Comment by m3taphysics** [comment]
Closing due to: https://github.com/decentraland/unity-renderer/pull/4186/files#diff-802617846bec437693e2fb4e126ce3ce8ff3777de96f79a31625d9b5a7ded0c8

Great minds @popuz

**Pull request closed without merging**